src/wp-includes/functions.php

```diff
@@ -7635,3 +7633,72 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param array     $attributes `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo       Optional. Prints to the page if true, returns the value otherwise.
+ * @param string    $content    Optional. Code to be wrapped around `<script>` tags.
```

Comment:
The signature of this function feels overly complex, and the logic slightly confusing. I think it would be more intuitive if `wp_script` was only used for script tags referencing a `src` and `wp_inline_script` was only used for inline scripts. For example, `wp_script` supporting a `$content` parameter feels unintuitive to me.

Maybe you implemented it the way you did so that you can reuse the logic, which makes sense. In that case I'd suggest though to introduce another somewhat "more private" function `wp_sanitize_script_attributes( $attributes, $double_quotes = true )` which could be used by both of them, returning the attribute string. `wp_script` and `wp_inline_script` could then be entirely separate, allowing for more intuitive signatures and clarified behavior.